gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -139,38 +190,90 @@ int main(int argc, char* argv[]) {
       continue;
     }
 
-    // Create output file path.
-    llvm::Twine output_file_path = llvm::Twine(output_dir)
-                                       .concat("/")
-                                       .concat(llvm::Twine(file_counter))
-                                       .concat(".test");
-
-    // Open output file for writing.
-    std::ofstream output_file(output_file_path.str());
-    if (!output_file.is_open()) {
-      std::cerr << "Failed to open output file: " << output_file_path.str()
-                << "\n";
-      return 4;
-    }
-
-    // Write the register definition lines into the output file.
-    output_file << register_defs_lines;
+    if (asm_output_enabled) {
+      // Create output file path.
+      llvm::Twine output_file_path = llvm::Twine(output_dir)
+                                         .concat("/")
+                                         .concat(llvm::Twine(file_counter))
+                                         .concat(".test");
 
-    // Multiple mappings can point to the same definition.
-    if (addrs->accessed_blocks.size() > 0) {
-      output_file << kMemDefPrefix << kMemNamePrefix << " " << addrs->block_size
-                  << " " << initial_mem_val_str << "\n";
-    }
-    for (const auto& addr : addrs->accessed_blocks) {
-      output_file << kMemMapPrefix << kMemNamePrefix << " " << std::dec << addr
+      // Open output file for writing.
+      std::ofstream output_file(output_file_path.str());
+      if (!output_file.is_open()) {
+        std::cerr << "Failed to open output file: " << output_file_path.str()
                   << "\n";
+        return 4;
+      }
+
+      // Write the register definition lines into the output file.
+      output_file << register_defs_lines;
+
+      // Multiple mappings can point to the same definition.
+      if (addrs->accessed_blocks.size() > 0) {
+        output_file << kMemDefPrefix << kMemNamePrefix << " "
+                    << addrs->block_size << " " << initial_mem_val_str << "\n";
+      }
+      for (const auto& addr : addrs->accessed_blocks) {
+        output_file << kMemMapPrefix << kMemNamePrefix << " " << std::dec
+                    << addr << "\n";
+      }
+
+      // Append disassembled instructions.
+      for (const auto& instr : proto->machine_instructions()) {
+        output_file << instr.assembly() << "\n";
+      }
     }
 
-    // Append disassembled instructions.
-    for (const auto& instr : proto->machine_instructions()) {
-      output_file << instr.assembly() << "\n";
+    if (json_output_enabled) {
+      llvm::json::Object current_snippet;
+
+      if (addrs->accessed_blocks.size() > 0) {
+        llvm::json::Array memory_definitions;
+        llvm::json::Object current_memory_definition;
+        current_memory_definition["Name"] = llvm::json::Value(kMemNamePrefix);
```

Comment:
IIUC the code right, `llvm::json::Value` has implicit conversions, and it should be possible to write just
```
current_memory_definition["Name"] = kMemNamePrefix;
```
and so on for the others.

Reply:
`kMemNamePrefix` is a `std::string_view` and it doesn't look like there's an explicit conversion for that one. So we either have to do `= std::string(kMemNamePrefix)` or `= llvm::json::Value(kMemNamePrefix)`.